ext/nnstreamer/tensor_filter/tensor_filter_python.c

```diff
@@ -0,0 +1,233 @@
+ * Copyright (C) 2019 Dongju Chae <dongju.chae@samsung.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ */
+/**
+ * @file	tensor_filter_python.c
+ * @date	10 Apr 2019
+ * @brief	python module for tensor_filter gstreamer plugin
+ * @see		http://github.com/nnsuite/nnstreamer
+ * @author	Dongju Chae <dongju.chae@samsung.com>
+ * @bug		No known bugs except for NYI items
+ */
+
+#define NO_IMPORT_ARRAY
+#include "tensor_filter_python_core.h"
+#include <nnstreamer_plugin_api_filter.h>
+#include <glib.h>
+#include <string.h>
+#include <nnstreamer_conf.h>
+
+/**
+ * @brief internal data of python
+ */
+struct _py_data
+{
+  void *py_private_data;
+};
+typedef struct _py_data py_data;
+
+static GstTensorFilterFramework *NNS_support_python;
+
+/**
+ * @brief Free privateData and move on.
+ */
+static void
+py_close (const GstTensorFilterProperties * prop, void **private_data)
+{
+  py_data *py;
+  py = *private_data;
+  py_core_delete (py->py_private_data);
+  g_free (py);
+  *private_data = NULL;
+}
+
+/**
+ * @brief Load python model
+ * @param prop: property of tensor_filter instance
+ * @param private_data : python plugin's private data
+ * @return 0 if successfully loaded. 1 if skipped (already loaded).
+ *        -1 if the object construction is failed.
+ *        -2 if the object initialization if failed
+ */
+static int
+py_loadScriptFile (const GstTensorFilterProperties * prop, void **private_data)
+{
+  /**
+   * prop->model_file contains the path of a python script
+   * prop->custom contains its arguments seperated by ' '
+   */
+  py_data *py;
+  if (*private_data != NULL) {
+    py = *private_data;
+    if (strcmp (prop->model_file, py_core_getScriptPath (py->py_private_data))) {
+      py_close (prop, private_data);
+    } else {
+      return 1;
+    }
+  }
+
+  py = g_new0 (py_data, 1); /** initialize py Fill Zero! */
+  *private_data = py;
+  py->py_private_data = py_core_new (prop->model_file, prop->custom_properties);
+
+  if (py->py_private_data) {
+    if (py_core_init (py->py_private_data, prop)) {
+      g_printerr ("failed to initailize the object: python");
+      return -2;
+    }
+    
+    /** check methods in python script */
+    cb_type type = py_core_getCbType (py->py_private_data);
+    switch (type) {
+      case CB_SETDIM:
+        NNS_support_python->getInputDimension = NULL;
+        NNS_support_python->getOutputDimension = NULL;
+        break;
+      case CB_GETDIM:
+        NNS_support_python->setInputDimension = NULL;
```

Comment:
What if the caller (nnstreamer application) loads up a model with SETDIM and then loads up a model with GETDIM? (All the three become "NULL"?)

Reply:
Got it. I will fix it.